src/plugins/data/server/index_patterns/fetcher/lib/field_capabilities/field_capabilities.ts

```diff
@@ -71,5 +72,11 @@ export async function getFieldCapabilities(
     )
     .map(mergeOverrides);
 
-  return sortBy(allFieldsUnsorted, 'name');
+  const filteredFields = !isEmpty(filters)
+    ? filters?.aggregatable === true
+      ? allFieldsUnsorted.filter((field) => field.aggregatable === true)
+      : allFieldsUnsorted
+    : allFieldsUnsorted;
```

Comment:
Do you only want this ability to filter for positive cases of aggregations? For example if someone wants to filter specifically on fields that aren't aggregatable will this work?

This test case doesn't pass at the moment:

```ts
    it('filters aggregatable fields that are false', async () => {
      const fieldsFromFieldCaps = [
        { name: 'foo', aggregatable: true },
        { name: 'bar', aggregatable: false },
      ];

      stubDeps({ fieldsFromFieldCaps });

      expect(
        await getFieldCapabilities(undefined, undefined, undefined, { aggregatable: false })
      ).toEqual([
        {
          name: 'bar',
          aggregatable: false,
          type: 'string',
          searchable: false,
          readFromDocValues: false,
        },
      ]);
    });
```

Which is fine if that's what you want.

Optional/nit you can reduce this to a single ternary operation like so if you want without an `isEmpty` if you like code golf ⛳  :-)

```ts
  const filteredFields =
    filters?.aggregatable != null
      ? allFieldsUnsorted.filter((field) => field.aggregatable === filters.aggregatable)
      : allFieldsUnsorted;
```

And that will also filter on both `true` and `false` aggregations.

Reply:
Good point! I had intended the former, but it kind of reads like the latter. I'll update it to handle both cases, thanks!

Reply:
@FrankHassanabad Updated!